contracts/PriceFeed/PriceFeed.sol

```diff
@@ -125,52 +160,82 @@ contract PriceFeed is Initializable, ProxyOwned {
+     * @param currencyKey The identifier of the currency.
+     * @return rate The current rate for the currency.
+     * @return time The timestamp when the rate was last updated.
+     */
     function rateAndUpdatedTime(bytes32 currencyKey) external view returns (uint rate, uint time) {
         RateAndUpdatedTime memory rateAndTime = _getRateAndUpdatedTime(currencyKey);
         return (rateAndTime.rate, rateAndTime.time);
     }
 
-    function removeFromArray(bytes32 entry, bytes32[] storage array) internal returns (bool) {
-        for (uint i = 0; i < array.length; i++) {
-            if (array[i] == entry) {
-                delete array[i];
-                array[i] = array[array.length - 1];
-                array.pop();
-                return true;
-            }
-        }
-        return false;
-    }
-
+    /**
+     * @notice Set the TWAP (Time-Weighted Average Price) calculation interval.
+     * @dev Only the owner can update the interval.
+     * @param _twapInterval The new TWAP interval in seconds.
+     */
     function setTwapInterval(int56 _twapInterval) external onlyOwner {
         twapInterval = _twapInterval;
         emit TwapIntervalChanged(_twapInterval);
     }
 
+    /**
+     * @notice Toggle the use of the last tick for TWAP calculation for a specified currency.
+     * @dev Only the owner can call this function.
+     * @param _currencyKey The identifier of the currency.
+     */
     function setLastTickForTWAP(bytes32 _currencyKey) external onlyOwner {
         useLastTickForTWAP[_currencyKey] = !useLastTickForTWAP[_currencyKey];
         emit LastTickForTWAPChanged(_currencyKey);
     }
 
+    /**
+     * @notice Set the address for the wrapped ETH (wETH) token.
+     * @dev Only the owner can update the wETH token address.
+     * @param token The address of the wETH token.
+     */
     function setWETH(address token) external onlyOwner {
         _wETH = token;
         emit AddressChangedwETH(token);
     }
 
+    /**
+     * @notice Set the address for the ETH token.
+     * @dev Only the owner can update the ETH token address.
+     * @param token The address of the ETH token.
+     */
     function setETH(address token) external onlyOwner {
         _ETH = token;
         emit AddressChangedETH(token);
     }
 
-    function setStaticPricePerAsset(bytes32 currencyKey, uint216 rate) external onlyOwner {
+    /**
+     * @notice Set a static price for a given asset.
+     * @dev Only the owner or whitelisted addresses can call this function. If the currency key does not exist in the list, it will be added.
+     * @param currencyKey The identifier of the asset.
+     * @param rate The static price to be set.
+     */
+    function setStaticPricePerAsset(bytes32 currencyKey, uint216 rate) external {
+        require(whitelistedAddresses[msg.sender] || msg.sender == owner, "Only whitelisted can set static price");
```

Comment:
I think whitelisted shouldnt be able to call this method, otherwise it makes no sense to have two methods for setting static price